keras/layers/preprocessing/text_vectorization.py

```diff
@@ -123,7 +123,8 @@ class TextVectorization(Layer):
             have its time dimension padded or truncated to exactly
             `output_sequence_length` values, resulting in a tensor of shape
             `(batch_size, output_sequence_length)` regardless of how many tokens
-            resulted from the splitting step. Defaults to `None`.
+            resulted from the splitting step. Defaults to `None`. If `ragged`
+            is `True` then `output_sequence_length` may truncate the output.
```

Comment:
Can you add a simple unit test to check that this behavior is verified?

Reply:
I am unable to pass this simple test:
```python
@pytest.mark.skipif(
        backend.backend() != "tensorflow", reason="Requires ragged tensors."
    )
    def test_ragged_tensor(self):
        layer = layers.TextVectorization(
            output_mode="int",
            vocabulary=["baz", "bar", "foo"],
            ragged=True,
        )
        input_data = [["foo baz baz bar"], ["foo baz bar"], ["foo baz"], ["foo"]]
        output = layer(input_data)
        assert isinstance(output, tf.RaggedTensor)
```

Fails with:
```bash
>       assert isinstance(output, tf.RaggedTensor)
E       AssertionError: assert False
E        +  where False = isinstance(<tf.Tensor: shape=(4, 4), dtype=int64, numpy=\narray([[4, 2, 2, 3],\n       [4, 2, 3, 0],\n       [4, 2, 0, 0],\n       [4, 0, 0, 0]], dtype=int64)>, <class 'tensorflow.python.ops.ragged.ragged_tensor.RaggedTensor'>)
E        +    where <class 'tensorflow.python.ops.ragged.ragged_tensor.RaggedTensor'> = tf.RaggedTensor

keras\layers\preprocessing\text_vectorization_test.py:120: AssertionError
=============================================== short test summary info ===============================================
FAILED keras/layers/preprocessing/text_vectorization_test.py::TextVectorizationTest::test_ragged_tensor - AssertionError: assert False
```

However the same test passes on Colab. Can you help?

Reply:
@fchollet please have a look

Reply:
It is looking to me like the layer as it is currently implemented is simply not wired to return ragged tensors.

Reply:
Address in #19028 that I started independently for another fix for Text Vectorization and also updated it to address issues here.

Reply:
Thank you @sampathweb